mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -305,6 +306,70 @@ void backupAccountWithTwoMailboxWithAnnotationsThenRestoringItInUser2AccountShou
         backupAccountWithTwoMailboxWithAnnotationsThenRestoringItInUser2AccountShouldRestoreAllElements();
     }
 
+    private void backupAccountWithOneMailboxAndTwoMessageThenRestoringItInUser2AccountShouldCreateOneMailboxWithTwoMessage() throws Exception {
+        createMailBoxWithMessages(sessionUser, MAILBOX_PATH_USER1_MAILBOX1, getMessage1AppendCommand(), getMessage2AppendCommand());
+
+        ByteArrayOutputStream destination = new ByteArrayOutputStream(BUFFER_SIZE);
+        backup.backupAccount(USER1, destination);
+
+        InputStream source = new ByteArrayInputStream(destination.toByteArray());
+        Mono.from(backup.restore(USER2, source)).block();
+
+        List<DefaultMailboxBackup.MailAccountContent> content = backup.getAccountContentForUser(sessionOtherUser);
+
+        assertThat(content).hasSize(1);
+        DefaultMailboxBackup.MailAccountContent mailAccountContent = content.get(0);
+        Mailbox mailbox = mailAccountContent.getMailboxWithAnnotations().mailbox;
+        assertThat(mailbox.getName()).isEqualTo(MAILBOX_1_NAME);
+        assertThat(mailAccountContent.getMessages().count()).isEqualTo(2);
+    }
+
+    @Disabled
+    @Test
+    void backupAccountWithOneMailboxAndTwoMessageThenRestoringItInUser2AccountShouldCreateOneMailboxWithTwoMessageInMemory() throws Exception {
+        backupAccountWithOneMailboxAndTwoMessageThenRestoringItInUser2AccountShouldCreateOneMailboxWithTwoMessage();
+    }
+    
+    @Disabled
+    @Test
+    void backupAccountWithOneMailboxAndTwoMessageThenRestoringItInUser2AccountShouldCreateOneMailboxWithTwoMessageInFile() throws Exception {
+        prepareForTestWithContentInTempFile();
+        backupAccountWithOneMailboxAndTwoMessageThenRestoringItInUser2AccountShouldCreateOneMailboxWithTwoMessage();
+    }
+
+    private void backupAccountWithOneMailboxWithAnnotationsAndTwoMessageThenRestoringItInUser2AccountShouldRestoreAllElements() throws Exception {
+        createMailBoxWithMessages(sessionUser, MAILBOX_PATH_USER1_MAILBOX1, getMessage1AppendCommand(), getMessage2AppendCommand());
+        mailboxManager.updateAnnotations(MAILBOX_PATH_USER1_MAILBOX1, sessionUser, WITH_ANNOTATION_1_AND_2);
+        ByteArrayOutputStream destination = new ByteArrayOutputStream(BUFFER_SIZE);
+        backup.backupAccount(USER1, destination);
+
+        InputStream source = new ByteArrayInputStream(destination.toByteArray());
+        Mono.from(backup.restore(USER2, source)).block();
+
+        List<DefaultMailboxBackup.MailAccountContent> content = backup.getAccountContentForUser(sessionOtherUser);
```

Comment:
the setup steps are similar among tests. And I find it's hard to ready a whole test, because it's big  & complex. I think you can extract some parts of your test to some method, and it can be reused an some where else. I suggest:

```java
@Test
void backUpShould...() {
     createUser1MailboxAndMessages();
     backupUser1ThenRestoreToUser2();

     assertThat(backup.getAccountContentForUser(user2Session))
             .hasSize(1)
             .allSatisfy(mailAccountContent -> {
                   ....
                    assertThat(mailbox.getName()).isEqualTo(MAILBOX_1_NAME);
                    assertThat(annotations).isEqualTo(WITH_ANNOTATION_1_AND_2);
                    assertThat(mailAccountContent.getMessages().count()).isEqualTo(2);
              });
}
```